### PR TITLE
switch to using Object.hashAll

### PR DESCRIPTION
**Pull request by devoncarew**
- switch to using Object.hashAll

I'm so-so on this change as the equals methods still use `ListEquality`, but the hash methods have switch to Object.hashAll.

### CODE REVIEW

**Comment by kevmoo** [comment]
Crap. `Object.hashAll` does not do DEEP equality, which we might need here. 🤷

**Comment by kevmoo** [comment]
Or maybe not. Just need to change the CI script to not be 2.12!

**Comment by devoncarew** [comment]
> Just need to change the CI script to not be 2.12!

updating...

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # 2.1.1-dev
 
 - Add markdown badges to the readme.
+- Switch to using `Object.hashAll` (from package:collection's
```
Comment (kevmoo) [inline CHANGELOG.md:4]: Meh...implementation detail. SDK constraint change LGTM, though

Comment (devoncarew) [reply CHANGELOG.md]: done

File: lib/src/version.dart [hunk 2]
```diff
@@ -175,8 +175,8 @@ class Version implements VersionConstraint, VersionRange {
       major ^
       minor ^
       patch ^
-      _equality.hash(preRelease) ^
-      _equality.hash(build);
+      Object.hashAll(preRelease) ^
```
Comment (kevmoo) [inline lib/src/version.dart:178]: ☝️ This should be `Object.hash(major, minor, patch, ...`

File: lib/src/version.dart [hunk 3]
```diff
@@ -172,9 +172,7 @@ class Version implements VersionConstraint, VersionRange {
 
   @override
   int get hashCode =>
-      major ^
-      minor ^
-      patch ^
+      Object.hash(major, minor, patch) ^
```
Comment (kevmoo) [inline lib/src/version.dart:175]: ditch the `^`!! Put the other things IN here.

File: lib/src/version.dart (hunk 2 above)
Comment (devoncarew) [reply lib/src/version.dart:176]: I really don't know whether `Object.hash` will treat `preRelease` - a List - as an object, and hash it directly, or know to treat it as a collection and hash the contents. I may just drop this PR as it's not an important change.

**Pull request closed without merging**